app/views/models/index.html.erb

```diff
@@ -2,32 +2,7 @@
 
 <% title_actions new_link(_("Create Model")) %>
 
-<table class="<%= table_css_classes 'table-fixed' %>">
-  <thead>
-    <tr>
-      <th class="col-md-4"><%= sort :name, :as => s_("Model|Name") %></th>
-      <th class="col-md-3"><%= sort :vendor_class, :as => _("Vendor Class") %></th>
-      <th class="col-md-3"><%= sort :hardware_model, :as => s_("Model|Hardware Model") %></th>
-      <th class="col-md-1"><%= _("Hosts") %></th>
-      <th><%= _('Actions') %></th>
-    </tr>
-  </thead>
-  <tbody>
-    <% @models.each do |model| %>
-      <tr>
-        <td class="ellipsis">
-          <%= link_to_if_authorized model.name,
-            hash_for_edit_model_path(:id => model).merge(:auth_object => model, :authorizer => authorizer) %></td>
-        <td class="ellipsis"><%= model.vendor_class %></td>
-        <td class="ellipsis"><%= model.hardware_model %></td>
-        <td class='ra'><%= link_to hosts_count[model], hosts_path(:search => "model = \"#{model}\"") %></td>
-        <td class="col-md-1">
-          <%= action_buttons(display_delete_if_authorized(hash_for_model_path(:id => model).
-            merge(:auth_object => model, :authorizer => authorizer),
-            :confirm => _("Delete %s?") % model.name)) %>
-        </td>
-      </tr>
-    <% end %>
-  </tbody>
-</table>
-<%= will_paginate_with_info @models %>
+<div id="models_table"></div>
+<%= mount_react_component("ModelsTable", "#models_table", {
```

Comment:
IMO we should use the component-wrapper only when we want to update those props with redux.
`data` prop became a standard for us to know those props came from server rendering and it feels useful to me.

Reply:
@amirfefer what are you thoughts ?
can you share about the `flat` option in `mount_react_component`  as we discussed offline ?

Reply:
you can use `{flattenData: true}` in the componentRegistery,
I do agree with @sharvit on this, but if we use a component both server and client rendering (like charts) IMO it would be better to use the `flattenData`